inst/tests/tests.Rraw

```diff
@@ -18111,3 +18111,15 @@ test(2238.9, NA %notin% c(1:5, NA), FALSE)
 
 # shift actionable error on matrix input #5287
 test(2239.1, shift(matrix(1:10, ncol = 1)), error="consider wrapping")
+
+# as.data.table should remove extra attributes from extended data.frames #5699
+x = data.frame(a=c(1,5,3), b=c(2,4,6))
+class(x) = c("tbl","data.frame")
+attr(x, "t1") = "a"
+as.data.frame.tbl = function(x) {
+  attr(x, "t1") = NULL
+  class(x) = "data.frame"
+  x
+}
+y = as.data.table(x)
+test(2240.1, attr(y, "t1", TRUE), NULL)
```

Comment:
this test happens to fail from R CMD check but not when doing interactive test from cc()

Reply:
I am missing something about S3 dispatch mechanics, but for whatever reason when run in batch dispatch to`as.data.frame.tbl()` doesn't happen.

It probably needs to be properly registered (e.g. with `registerS3method()`), but I'm a bit loath to mess with the session S3 dispatch table like that.

I amended the test to only check that the `as.data.table()` and `as.data.frame()` behavior with respect to that attribute agree.